domain/src/main/java/org/oppia/android/domain/classroom/ClassroomController.kt

```diff
@@ -0,0 +1,386 @@
+      .setTotalChapterCount(totalChapterCount)
+      .setTopicThumbnail(createTopicThumbnailFromJson(jsonObject))
+      .setTopicPlayAvailability(topicPlayAvailability)
+      .setFirstStoryId(firstStoryId)
+      .build()
+  }
+}
+
+/** Creates a [LessonThumbnail] from a classroomJsonObject. */
+internal fun createClassroomThumbnailFromJson(classroomJsonObject: JSONObject): LessonThumbnail {
+  val classroomId = classroomJsonObject.optString("classroom_id")
+  val thumbnailBgColor = classroomJsonObject.optString("thumbnail_bg_color")
+  val thumbnailFilename = classroomJsonObject.optString("thumbnail_filename")
+  return if (thumbnailFilename.isNotNullOrEmpty() && thumbnailBgColor.isNotNullOrEmpty()) {
+    LessonThumbnail.newBuilder()
+      .setThumbnailFilename(thumbnailFilename)
+      .setBackgroundColorRgb(Color.parseColor(thumbnailBgColor))
+      .build()
+  } else if (CLASSROOM_THUMBNAILS.containsKey(classroomId)) {
+    CLASSROOM_THUMBNAILS.getValue(classroomId)
+  } else {
+    createDefaultClassroomThumbnail()
+  }
+}
+
+/** Creates a [LessonThumbnail] from a classroom proto. */
+internal fun createClassroomThumbnailFromProto(
+  classroomId: String,
+  lessonThumbnail: LessonThumbnail
+): LessonThumbnail {
+  val thumbnailFilename = lessonThumbnail.thumbnailFilename
+  return when {
+    thumbnailFilename.isNotNullOrEmpty() -> lessonThumbnail
+    CLASSROOM_THUMBNAILS.containsKey(classroomId) -> CLASSROOM_THUMBNAILS.getValue(classroomId)
+    else -> createDefaultClassroomThumbnail()
+  }
+}
+
+/** Creates a default [LessonThumbnail]. */
+internal fun createDefaultClassroomThumbnail(): LessonThumbnail {
+  return LessonThumbnail.newBuilder()
+    .setThumbnailGraphic(LessonThumbnailGraphic.MATHS_CLASSROOM)
+    .setBackgroundColorRgb(Color.parseColor(CLASSROOM_BG_COLOR))
+    .build()
+}
+
+/** Creates a [LessonThumbnail] for [TEST_CLASSROOM_ID_0]. */
+internal fun createClassroomThumbnail0(): LessonThumbnail {
+  return LessonThumbnail.newBuilder()
+    .setThumbnailGraphic(LessonThumbnailGraphic.SCIENCE_CLASSROOM)
+    .setBackgroundColorRgb(Color.parseColor(CLASSROOM_BG_COLOR))
+    .build()
+}
+
+/** Creates a [LessonThumbnail] for [TEST_CLASSROOM_ID_1]. */
+internal fun createClassroomThumbnail1(): LessonThumbnail {
+  return LessonThumbnail.newBuilder()
+    .setThumbnailGraphic(LessonThumbnailGraphic.MATHS_CLASSROOM)
+    .setBackgroundColorRgb(Color.parseColor(CLASSROOM_BG_COLOR))
+    .build()
+}
+
+/** Creates a [LessonThumbnail] for [TEST_CLASSROOM_ID_2]. */
+internal fun createClassroomThumbnail2(): LessonThumbnail {
+  return LessonThumbnail.newBuilder()
+    .setThumbnailGraphic(LessonThumbnailGraphic.ENGLISH_CLASSROOM)
+    .setBackgroundColorRgb(Color.parseColor(CLASSROOM_BG_COLOR))
+    .build()
+}
+
+private fun String?.isNullOrEmpty(): Boolean = this == null || this.isEmpty() || this == "null"
+
+private fun String?.isNotNullOrEmpty(): Boolean = !this.isNullOrEmpty()
```

Comment:
I think this can be simplified to this: 
```suggestion
private fun String?.isNotNullOrEmpty(): Boolean = !this.isNullOrBlank() || this != "null"
```

Reply:
Done.